.travis.yml

```diff
@@ -1,31 +1,63 @@
 language: ruby
+
 sudo: false
 dist: trusty
+
 rvm:
   - 2.3.0
+
+# Allow Emacs snapshot builds to fail and don't wait for it
 matrix:
   fast_finish: true
   allow_failures:
     - env: EMACS_VERSION=snapshot
+
+# Cache stable Emacs binaries and packages
 cache:
-  - bundler
-  - apt: true
-  - directories:
+  bundler: true
+  apt: true
+  timeout: 691200
+  directories:
       - "$HOME/emacs"
+      - "$HOME/.emacs.d"
+      - vendor/bundle
+
+git:
+  depth: 1
+
 env:
   matrix:
+    - EMACS_VERSION=24.3
+    - EMACS_VERSION=24.4
     - EMACS_VERSION=24.5
-    - EMACS_VERSION=25.1-rc2
     - EMACS_VERSION=25.1
     - EMACS_VERSION=25.2
     - EMACS_VERSION=snapshot
     - TEXINFO_VERSION=6.1
+  global:
+    # Maximum number of executing procesors
+    - MAKE_JOBS=$(getconf _NPROCESSORS_ONLN)
+    # To cache vendor/bundle
+    - BUNDLE_PATH=vendor/bundle
+    # Setting up the parallel Install
+    - BUNDLE_JOBS=$(getconf _NPROCESSORS_ONLN)
+
 before_install:
+  # Configure $PATH: Executables are installed to $HOME/bin
   - export PATH="$HOME/bin:$PATH"
+
+install:
+  - bundle install --deployment # To cache vendor/bundle
+
+before_script:
+  # Make sure the exact Emacs version can be found in the build output
+  - $(command -v emacs) --version | head -1
+
 script:
-  - if [[ -n $EMACS_VERSION ]]; then make -f emacs-travis.mk install_emacs; fi
-  - if [[ -n $TEXINFO_VERSION ]]; then make -f emacs-travis.mk install_texinfo; fi
+  - '[[ ! -n $EMACS_VERSION ]] || make -s -f emacs-travis.mk install_emacs'
+  - '[[ ! -n $TEXINFO_VERSION ]] || make -s -f emacs-travis.mk install_texinfo'
```

Comment:
I'm not sure this is more readable.  I liked the `if then fi` version more.

A question about making `make` silent with `-s`: does it still report error messages if a command fails?